debugging/coreclr/dockerClient.ts

```diff
@@ -0,0 +1,194 @@
+
+        if (!imageId) {
+            throw new Error('The Docker image was built successfully but the image ID could not be retrieved.');
+        }
+
+        return imageId;
+    }
+
+    public async inspectObject(nameOrId: string, options?: DockerInspectObjectOptions): Promise<string | undefined> {
+        let command = 'docker inspect';
+
+        if (options && options.format) {
+            command += ` \"--format=${options.format}\"`;
+        }
+
+        command += ` ${nameOrId}`;
+
+        try {
+            const output = await this.processProvider.exec(command, {});
+
+            return output.stdout;
+        } catch {
+            // Failure (typically) means the object wasn't found...
+            return undefined;
+        }
+    }
+
+    public async listContainers(options?: DockerContainersListOptions): Promise<string> {
+        let command = 'docker ps -a';
+
+        if (options && options.format) {
+            command += ` \"--format=${options.format}\"`;
+        }
+
+        const output = await this.processProvider.exec(command, {});
+
+        return output.stdout;
+    }
+
+    public matchId(id1: string, id2: string): boolean {
+        const validateArgument =
+            id => {
+                if (id === undefined || id1.length < 12) {
+                    throw new Error(`'${id}' must be defined and at least 12 characters.`)
+                }
+            };
+
+        validateArgument(id1);
+        validateArgument(id2);
+
+        return id1.length < id2.length
+            ? id2.startsWith(id1)
+            : id1.startsWith(id2);
+    }
+
+    public async removeContainer(containerNameOrId: string, options?: DockerContainerRemoveOptions): Promise<void> {
+        let command = 'docker rm';
+
+        if (options && options.force) {
+            command += ' --force';
+        }
+
+        command += ` ${containerNameOrId}`;
+
+        await this.processProvider.exec(command, {});
+    }
+
+    public async runContainer(imageTagOrId: string, options?: DockerRunContainerOptions): Promise<string> {
+        let command = 'docker run -dt -P';
+
+        if (options && options.containerName) {
+            command += ` --name ${options.containerName}`;
+        }
+
+        if (options && options.volumes) {
+            command += ' ' + options.volumes.map(volume => `-v \"${volume.localPath}:${volume.containerPath}${volume.permissions ? ':' + volume.permissions : ''}\"`).join(' ');
+        }
+
+        if (options && options.entrypoint) {
+            command += ` --entrypoint ${options.entrypoint}`;
+        }
+
+        command += ' ' + imageTagOrId;
+
+        if (options && options.command) {
+            command += ' ' + options.command;
+        }
+
+        const result = await this.processProvider.exec(command, {});
+
+        const containerId = result.stdout.substr(0, result.stdout.length - 1 /* Exclude trailing <CR>. */);
```

Comment:
And if this is Windows?  CRLF?

Reply:
Fair point; updated to use `trim()` instead of making assumptions about exact number(s) of whitespace.  While Docker explicitly states the value in the output is the container ID, it doesn't appear to document their container ID format, so a regex here would still be relying on an assumption.

Reply:
trim() works